server/events/project_command_builder.go

```diff
@@ -347,7 +347,17 @@ func (p *DefaultProjectCommandBuilder) buildAllCommandsByCfg(ctx *command.Contex
 
 	if p.SkipCloneNoChanges && p.VCSClient.SupportsSingleFileDownload(ctx.Pull.BaseRepo) {
 		repoCfgFile := p.GlobalCfg.RepoConfigFile(ctx.Pull.BaseRepo.ID())
-		hasRepoCfg, repoCfgData, err := p.VCSClient.GetFileContent(ctx.Log, ctx.Pull, repoCfgFile)
+
```

Comment:
Can you add a test for this new code?

Reply:
I just added a test case, regenerated the `mock_client.go` file with pegomock, and fixed the issue w/ the DCO status check on my first commit. Should be good for another look 👍

Reply:
I just added a test case, regenerated the mock_client.go file with pegomock, and fixed the issue w/ the DCO status check on my first commit.

The test also made it clear I could simplify some of the logic to just always pass the `HeadRepo` to `GetFileContent`. With that being the case, maybe I should just revert the interface modifications and add the `HeadRepo` to `models.PullRequest`? Not entirely sure how feasible that is

https://github.com/runatlantis/atlantis/blob/22b37512ae4205df901e8b8a3746a8ada4b52c50/server/events/models/models.go#L156-L181